mmv1/third_party/terraform/services/cloudquotas/resource_cloud_quotas_quota_preference_test.go

```diff
@@ -0,0 +1,292 @@
+	t.Parallel()
+
+	context := map[string]interface{}{
+		"org_id":          envvar.GetTestOrgFromEnv(t),
+		"billing_account": envvar.GetTestBillingAccountFromEnv(t),
+		"random_suffix":   acctest.RandString(t, 10),
+	}
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		ExternalProviders: map[string]resource.ExternalProvider{
+			"time": {},
+		},
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCloudQuotasQuotaPreference_cloudquotasQuotaPreferenceBasicExample_basic(context),
+			},
+			{
+				ResourceName:            "google_cloud_quotas_quota_preference.my_preference",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"parent", "quota_preference_id", "allow_missing", "validate_only", "ignore_safety_checks", "contact_email"},
+			},
+			{
+				Config: testAccCloudQuotasQuotaPreference_cloudquotasQuotaPreferenceBasicExample_increaseQuota(context),
+			},
+			{
+				ResourceName:            "google_cloud_quotas_quota_preference.my_preference",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ExpectNonEmptyPlan:      true,
+				ImportStateVerifyIgnore: []string{"parent", "quota_preference_id", "allow_missing", "validate_only", "ignore_safety_checks", "contact_email", "justification", "quota_config.0.annotations"},
+			},
+			{
+				Config: testAccCloudQuotasQuotaPreference_cloudquotasQuotaPreferenceBasicExample_decreaseQuota(context),
+			},
+			{
+				ResourceName:            "google_cloud_quotas_quota_preference.my_preference",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"parent", "quota_preference_id", "allow_missing", "validate_only", "ignore_safety_checks", "contact_email", "justification"},
+			},
+			{
+				Config: testAccCloudQuotasQuotaPreference_cloudquotasQuotaPreferenceBasicExample_upsert(context),
+			},
+			{
+				ResourceName:            "google_cloud_quotas_quota_preference.my_preference",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"parent", "quota_preference_id", "allow_missing", "validate_only", "ignore_safety_checks", "contact_email", "justification"},
+			},
+		},
+	})
+}
+
+func testAccCloudQuotasQuotaPreference_cloudquotasQuotaPreferenceBasicExample_basic(context map[string]interface{}) string {
+	return acctest.Nprintf(`
+		resource "google_project" "new_project" {
+			project_id 		= "tf-test%{random_suffix}"
+			name       		= "tf-test%{random_suffix}"
+			org_id			= "%{org_id}"
+			billing_account	= "%{billing_account}"
+		}
+
+		resource "google_project_iam_binding" "project_iam" {
+			project 	= google_project.new_project.project_id
+			role    	= "roles/cloudquotas.admin"
+			members 	= [
+				"user:testinguser2@google.com"
```

Comment:
I think you will need to use the configured service account (which will be our test runner in CI environments). We just came across a similar case, so if you have issues setting up this permission, we can consider alternative approaches.

Reply:
Previously our backend is checking permission for the `contact_email` field, and the block grants essential permissions to the email account. We've removed our IAM checking mechanism and the change is now live in prod, so I've removed the block.

Looking into the test report, it seems the API is complaining about decreasing the quota by too much. I've added a parameter to force the change.